Geometry_Engine/Query/Centroid.cs

```diff
@@ -26,16 +26,44 @@
 using System;
 using BH.Engine.Reflection;
 using BH.oM.Reflection.Attributes;
+using BH.oM.Geometry.ShapeProfiles;
+using BH.oM.Base;
 
 namespace BH.Engine.Geometry
 {
     public static partial class Query
     {
+        /***************************************************/
+        /**** Public Methods - Misc                     ****/
+        /***************************************************/
+
+        public static Point Centroid(this PlanarSurface surface, double tolerance = Tolerance.Distance)
+        {
+            return Centroid(new List<ICurve> { surface.ExternalBoundary }, surface.InternalBoundaries, tolerance);
+        }
+
+        /***************************************************/
+
+
+        public static Point Centroid(this IProfile profile, double tolerance = Tolerance.Distance)
+        {
+            return Centroid(profile.Edges, new List<ICurve>(), tolerance);
+        }
+
+        /***************************************************/
+
+        //TODO: Move to structure_engine probably
+
+        //public static Point Centroid(this Panel panel, double tolerance = Tolerance.Distance)
+        //{
+        //    return Centroid(panel.ExternalEdges.Select(e => e.Curve).ToList(), panel.Openings.Select(o => o.Edges.Select(e => e.Curve)).SelectMany(o => o).ToList(), tolerance);
+        //}
+
```

Comment:
Agree, this should not be in here. Should also probably make sure this works on IPanel/IElement2d